### PR TITLE
Fix #45

### LINKED ISSUE #45: Code browsing
It would be quite convenient to be able to quickly browse the code to fully understand the experiment.
Currently, you have to first download the file before you can open it. It would be more convenient if it was additionally possible to just open the file in a new tab without having it to download it first.

### PR DESCRIPTION
**Pull request by kilsenp**
There were two bugs:
- I had an empty file, which was saved by sacred but no chunk in fs.chunks was created.
- Files in folders were not displayed properly. This was because in the database the full path was saved, but sacred saved the source files the relative path.
The filename in creation in L. 55 in sourceFilesView.js did not take files in folders into account. By using the file id instead of the file name as a key, it becomes easier to associate the source file with the file in the database.

This pull request makes the code viewing more robust.
- Checks if chunk exists.
- Uses file ID instead of name: Sacred saves the relative path or source files but the DB saves the absolute path. Using the file id is unambiguous.

### CODE REVIEW

File: web/src/components/SourceFilesView/sourceFilesView.js [hunk 1]
```diff
@@ -52,13 +52,21 @@ class SourceFilesView extends Component {
...
-          files[fileName] = {
+          var dataChunk;
+          if (file.chunk.length == 1) {
+            dataChunk = file.chunk[0].data;
+          } else if (file.chunk.length > 1) {
+            // Does this ever happen?
+            dataChunk = "";
+          } else {
+            dataChunk = "";
+          }
```
Comment (vivekratnavel) [inline web/src/components/SourceFilesView/sourceFilesView.js]: I don't think this part is required. Let's just leave data as `data: file.chunk[0].data` for now

Comment (kilsenp) [reply web/src/components/SourceFilesView/sourceFilesView.js]: This caused the error for me.
_I had an empty file, which was saved by sacred but no chunk in fs.chunks was created._
Obviously, the else if is not necessary right now, but I felt like it's clearer. But I can remove the 'else if' if you want me to

**Comment by kilsenp** [comment]
I am not quite sure, why I had to update snapshot in the last commit changed. 

    - Snapshot
    + Received
    
    @@ -151,8 +151,6 @@
           "variable": Object {
             "color": "#7ec699",
           },
         }
       }
    ->
    -  }qXresultqXHello world!qs.
    -</SyntaxHighlighter>
    +/>

**Comment by vivekratnavel** [comment]
@kilsenp I figured why snapshot test was failing.
`filePreview.test.js` uses the following sourceFiles to run tests - 
```
const sourceFiles = {
    "hello_world.py": {
      data: "SGVsbG8gV29ybGQh"
    },
    "result.pickle": {
      data: "gAN9cQBYBgAAAHJlc3VsdHEBWAwAAABIZWxsbyB3b3JsZCFxAnMu"
    }
  };
```
The keys used here are `fileNames`. We should change these keys to `fileId`s and modify fileIds appropriately in the tests instead of updating the snapshot.

**Comment by kilsenp** [comment]
Good catch! It uncovered another error which also should be fixed now.

**Pull request merged**